packages/stack/proxy/src/rpc-manager.ts

```diff
@@ -0,0 +1,193 @@
+import { __ } from "embark-i18n";
+import { Events } from "embark-core";
+import { Account, Callback, Embark, EmbarkEvents } from "embark-core";
+import { Logger } from 'embark-logger';
+import { AccountParser, dappPath } from "embark-utils";
+import { ProxyRequestParams, ProxyResponseParams } from '.';
+import Web3 from "web3";
+
+export type RpcRequestInterceptor<T> = (params: ProxyRequestParams<T>) => ProxyRequestParams<T>;
+export type RpcResponseInterceptor<T, R> = (params: ProxyRequestParams<T> | ProxyResponseParams<T, R>) => ProxyResponseParams<T, R>;
+
+interface RegistrationOptions {
+  priority: number;
+}
+
+interface RpcRegistration {
+  filter: string | string[] | RegExp;
+  options: RegistrationOptions;
+}
+
+interface RpcRequestRegistration<T> extends RpcRegistration {
+  interceptor: RpcRequestInterceptor<T>;
+}
+
+interface RpcResponseRegistration<T, R> extends RpcRegistration {
+  interceptor: RpcResponseInterceptor<T, R>;
+}
+
+export default class RpcManager {
+
+  private _web3: Web3 | null = null;
+
+  private logger: Logger;
+
+  private events: EmbarkEvents;
+
+  public _accounts: Account[] | null = null;
+
+  public _nodeAccounts: string[] | null = null;
+
+  private requestInterceptors: Array<RpcRequestRegistration<any>> = [];
+
+  private responseInterceptors: Array<RpcResponseRegistration<any, any>> = [];
+
+  constructor(private readonly embark: Embark) {
+    this.events = embark.events;
+    this.logger = embark.logger;
+  }
+
+  protected get web3() {
+    return (async () => {
+      if (!this._web3) {
+        await this.events.request2("blockchain:started");
+        // get connection directly to the node
+        const provider = await this.events.request2("blockchain:node:provider", "ethereum");
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
+
+  private get nodeAccounts() {
+    return (async () => {
+      if (!this._nodeAccounts) {
+        const web3 = await this.web3;
+        this._nodeAccounts = await web3.eth.getAccounts();
+      }
+      return this._nodeAccounts || [];
+    })();
+  }
+
+  private get accounts() {
+    return (async () => {
+      if (!this._accounts) {
+        const web3 = await this.web3;
+        const nodeAccounts = await this.nodeAccounts;
+        this._accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, nodeAccounts);
+      }
+      return this._accounts || [];
+    })();
+  }
```

Comment:
I'm fairly certain these are not needed here. They should only be needed in the `RpcInterceptor` base class.